hw/ip/csrng/data/csrng.hjson

```diff
@@ -95,18 +95,66 @@
     { name: "INTERSIG.MUBI"
       desc: "OTP signal used to enable software access to registers."
     }
-    { name: "FSM.SPARSE"
-      desc: "Sparse state machine implementation."
+    { name: "MAIN_SM.FSM.SPARSE"
+      desc: "The CSRNG main state machine uses a sparse state encoding."
     }
-    { name: "CTR.REDUN"
-      desc: "Counter hardening for generate command counter."
+    { name: "UPDATE.FSM.SPARSE"
+      desc: "The CSRNG update state machine uses a sparse state encoding."
+    }
+    { name: "BLK_ENC.FSM.SPARSE"
+      desc: "The CSRNG block encrypt state machine uses a sparse state encoding."
+    }
+    { name: "OUTBLK.FSM.SPARSE"
+      desc: "The CSRNG block output state machine uses a sparse state encoding."
+    }
+    { name: "GEN_CMD.CTR.REDUN"
+     desc: "The generate command uses a counter that is protected by a second counter that counts in the opposite direction."
+    }
+    { name: "DRBG.CTR.REDUN"
+     desc: "The ctr_drbg algorthm uses a counter that is protected by a second counter that counts in the opposite direction."
+    }
+    { name: "CTRL.MUBI"
+      desc: "Multi-bit field used for selection control."
+    }
+    { name: "MAIN_SM.CTR.LOCAL_ESC"
+      desc: "A mismatch detected inside any CSRNG counter moves the  main state machine into a terminal error state."
+    }
+    { name: "CONSTANTS.LC_GATED"
+      desc: "Seed diversification based on the lifecycle state."
+    }
+    { name: "FSM.LC_GATED"
+      desc: "Debug state machines are gated with lifecycle state."
     }
     { name: "SW_GENBITS.BUS.CONSISTENCY"
       desc: "Comparison on successive bus values for genbits returned on the software channel."
     }
     { name: "TILE_LINK.BUS.INTEGRITY"
       desc: "Tilelink end-to-end bus integrity scheme."
     }
+    { name: "AES_CIPHER.FSM.SPARSE",
```

Comment:
@vogelpi @cdgori @tjaychen 
What is your view on this?
Should we replicate these labels here, or would it be ok to omit them, since they are presumably already listed on the AES side?

Reply:
i'm okay with omitting them, but would this cause issues with the regtool check?
I'm hoping it doesn't.  Does the regtool check look at the rtl first then look for a matching hjson entry?

or does it look for the hjson entry first and find if it's there in the rtl? (probably the latter?)

Reply:
It does both and ensures that (1) each hjson entry has at least one matching entry in the RTL and (2) each entry in the RTL corresponds to an hjson entry.

Reply:
I think it makes sense to list these labels here as the AES core and it's countermeasures (similar to the SHA3 core used inside entropy source) are parametrized. In particular, inside CSRNG and entropy source we don't enable the masking for both these cores. And by not listing the masking countermeasure here we make this more clear.

Reply:
What @rswarbrick said seems the lightest-weight way to deal with this.  And as @vogelpi points out, by at least enumerating what we expect here we can highlight that we aren't using the masked version of the AES.

(I can't think of a good way to incorporate all of AES's countermeasures "by reference" because which ones you have depend on the specific parameters for the instance.  If someone can, that would cut down on the bookkeeping/consistency-keeping even further.)

Reply:
I have added and remove and added these already. Will wait for consensus now.

Reply:
I should probably also note that the regtool currently only checks RTL files local to this IP, which means that cross-IP label references are not possible right now (for that we would have to beef up the checking mechanism to leverage fusesoc in order to gather all relevant files).

I would keep this as simple as possible.
Can we maybe just add a one general item along these lines:
```
name: `AES_CIPHER.LOGIC.CM`
desc: ```
      This module uses an unmasked AES implementation (maybe extend description here a bit).
      See the AES module documentation for AES-specific countermeasures. 
      ```

Otherwise it becomes really difficult to maintain consistency accross the descriptions of the individual items.

Reply:
yeah i agree, broad category is best.
The only reason I think we "might" want to handle (in this case), is because AES has a lot of parameter hardening options. And CSRNG if I remember does not make use of most of them.

Reply:
The proposal of @msfschaffner sounds good to me, too.